libs/auth/src/java_card.ts

```diff
@@ -236,20 +284,66 @@ export class JavaCard implements Card {
     return Promise.resolve();
   }
 
-  readData(): Promise<Buffer> {
-    return Promise.resolve(Buffer.from([]));
+  unprogram(): Promise<void> {
+    return Promise.resolve();
   }
 
-  writeData(): Promise<void> {
-    return Promise.resolve();
+  async readData(): Promise<Buffer> {
+    const chunks: Buffer[] = [];
+    for (const objectId of GENERIC_STORAGE_SPACE.OBJECT_IDS) {
+      let response: Buffer;
+      try {
+        response = await this.cardReader.transmit(
+          new CardCommand({
+            ins: GET_DATA.INS,
+            p1: GET_DATA.P1,
+            p2: GET_DATA.P2,
+            data: constructTlv(GET_DATA.TAG_LIST_TAG, objectId),
+          })
+        );
+      } catch (error) {
+        if (
+          error instanceof ResponseApduError &&
+          isFileNotFoundStatusWord(error.statusWord())
+        ) {
+          chunks.push(Buffer.from([]));
+          continue;
```

Comment:
When would this case occur? Why is it skippable?

Reply:
Explained in https://github.com/votingworks/vxsuite/pull/3045/commits/7d86f4f78d9e3f8c98acc57d9e27eb9f5fa15092